### ISSUE TITLE
Check if training data matches MITIE requirements

### ISSUE THREAD

**Issue by derhagen** [golastmile/rasa_nlu#113]
With my training set, I often get difficult to understand error messages from MITIE. It would be nice if rasa checked the set before giving it to MITIE and provided understandable error messages.

**Comment by amn41** [comment]
Hi, thanks for creating an issue. Could you please provide a specific example?

**Comment by derhagen** [comment]
Sure!
...
  File "build/bdist.macosx-10.7-x86_64/egg/rasa_nlu/trainers/mitie_trainer.py", line 43, in train_entity_extractor
  File "/Users/XXX/miniconda3/envs/rasa/lib/python2.7/site-packages/mitie/mitie.py", line 461, in add_entity
    raise Exception("Invalid range given to ner_training_instance.overlaps_any_entity().  "
Exception: Invalid range given to ner_training_instance.overlaps_any_entity().  It overlaps an entity given to a previous call to add_entity().

**Comment by amn41** [comment]
Ok, looks like you have overlapping entities. MITIE doesn't allow that but spaCy does. Could you please add your training data to confirm? We can certainly add a warning that this isn't supported with the mitie backend.

**Comment by derhagen** [comment]
Thanks! Sent you a mail with the .json

**Comment by derhagen** [comment]
I tested my data for overlapping entities using this script, but I could not find any.
```
import json
from pprint import print

with open('hhb_m_cropped.json') as data_file:
    data = json.load(data_file)

examples = data["rasa_nlu_data"]["common_examples"]

for training_example in examples:
    entities = training_example['entities']
    if len(entities) > 1:
        sorted_entities = sorted(entities, key=lambda x: x['start'])
        for first, second in zip(sorted_entities, sorted_entities[1:]):
            if first['end'] > second['start']:
                pprint(training_example)
```

**Comment by amn41** [comment]
Nope - thanks for sharing the data. It's actually an edge case in the algorithm for finding the entities in the string :) Adding a bug label

**Comment by amn41** [comment]
I just pushed a commit bf466db246a065c10968989f0933bfb2c04bd6da which should address this issue.

**Comment by karnigili** [comment]
hi seems that I have the same issue - Exception("Invalid range given to ner_training_instance.overlaps_any_entity()"). as well could not fins any overlaps.

**Comment by ThejanW** [comment]
I have the same issue - Exception("Invalid range given to ner_training_instance.overlaps_any_entity()"). Can't find any overlapping entities.

**Comment by wrathagom** [comment]
@ThejanW please create a new issue and share your training data with us in a gist if possible.
